elevation_mapping:
  ros__parameters:
 
    # GENERAL
    elevation_map_topic: "/elevation_map"
    map_frame_id: "lidar_odom"
    robot_base_frame_id: "base_link"
    sensor_frame_id: "livox_frame"
    robot_odom_topic: "/Odometry"
    robot_odom_cache_size: 200
    robot_pose_with_covariance_topic: "base_link_pose"
    # LIDAR
    scanning_duration: 0.1
    input_sources: ["lidar"] # A name to identify the input source
    lidar:
      type: pointcloud # Supported types: pointcloud
      topic: /filtered_points
      queue_size: 10
      publish_on_update: true # Whether to publish the elevation map after a callback from this source.
      sensor_processor:
          type: laser
          min_radius: 0.018
          beam_angle: 0.0006
          beam_constant: 0.0015
          ignore_points_above: 3.0
          ignore_points_below: -3.0
 
    # MAP ORIGIN
    track_point_frame_id: "base_link"
    track_point_x: 0.0
    track_point_y: 0.0
    track_point_z: -0.197
 
    # MAP PARAMETERS
    length_in_x: 10.0
    length_in_y: 10.0
    resolution: 0.05   # in m/cell
    min_update_rate: 10.0
    min_variance: 0.000009
    max_variance: 0.01
    mahalanobis_distance_threshold: 3.5
    multi_height_noise: 0.009 #0.009
 
    # MAP CLEANUP
    fused_map_publishing_rate: 1.0
    enable_visibility_cleanup: true
    visibility_cleanup_rate: 1.0 #0.1 
    enable_continuous_cleanup: false

    # INITIALIZATION
    initialize_elevation_map: false
    initialization_method: 0
    length_in_x_init_submap: 10.2
    length_in_y_init_submap: 10.8
    init_submap_height_offset: 0.0
    init_submap_variance: 0.01
    target_frame_init_submap: "/base_link"

    # OTHER PARAMETERS
    underlying_map_topic: ""
    increase_height_alpha: 0.0
    masked_replace_service_mask_layer_name: "mask"
    
    enable_circular_clear: true
    circular_clear_radius: 1.0  # Example radius in meters
    circular_clear_frequency: 1.0  # Frequency in Hz (5 seconds interval)
